### PR TITLE
[Feature] JS: Addinig Open Layers format for reading WFS capabilities data

### PR DESCRIPTION
**Pull request by rldhont**
Adding Open Layers format for reading WFS capabilities data provided by [ol-wfs-capabilities](https://github.com/GastonZalba/ol-wfs-capabilities) package.

### CODE REVIEW

File: assets/src/modules/Config.js [hunk 1]
```diff
@@ -23,17 +23,27 @@ import { DatavizOptionsConfig, DatavizLayersConfig } from './config/Dataviz.js';
 import { buildLayerTreeConfig, LayerTreeGroupConfig } from './config/LayerTree.js';
 import { buildLayersOrder } from './config/LayersOrder.js';
 
+/**
+ * @typedef WfsFeatureType
+ * @type {object}
+ * @property {string} Name  - The layer typename.
+ * @property {string} Title - The layer title.
+ * @property {string} SRS   - The layer CRS code.
```
Comment (Gustry) [inline assets/src/modules/Config.js:31]: SRS or CRS ? ;-)

File: assets/src/modules/Config.js [hunk 2]
```diff
@@ -23,17 +23,27 @@ import { DatavizOptionsConfig, DatavizLayersConfig } from './config/Dataviz.js';
...
  * @name Config
  */
 export class Config {
 
     /**
-     * @param {object} cfg - the lizmap config object
-     * @param {object} wmsCapabilities - the wms capabilities
+     * @param {object} cfg               - the lizmap config object
+     * @param {object} wmsCapabilities   - the wms capabilities
+     * @param {object} [wfsCapabilities] - the wms capabilities
```
Comment (Gustry) [inline assets/src/modules/Config.js:44]:
```suggestion
     * @param {object} wmsCapabilities   - the WMS capabilities
     * @param {object} [wfsCapabilities] - the WFS capabilities
```

File: assets/src/modules/Config.js [hunk 3]
```diff
@@ -216,6 +230,42 @@ export class Config {
         return [...this._layersOrder];
     }
 
+    /**
+     * The list of format for file export
+     * @type {string[]}
```
Comment (Gustry) [inline assets/src/modules/Config.js:235]:
```suggestion
     * @returns {string[]}
```
?

File: assets/src/modules/Config.js [hunk 4]
```diff
@@ -216,6 +230,42 @@ export class Config {
...
+                continue;
+            }
+            return request.ResultFormat
+        }
+        return formats;
+    }
+
+    /**
+     * The list of WFS feature type
+     * @type {WfsFeatureType[]}
```
Comment (Gustry) [inline assets/src/modules/Config.js:254]: Same kind of return ?

**Review by Gustry — COMMENTED** [review]
Minor review

File: assets/src/modules/Config.js (hunk 3 above)
Comment (rldhont) [reply assets/src/modules/Config.js:235]: It is a getter not a method. In this case `@returns` is not valid.

File: assets/src/modules/Config.js (hunk 4 above)
Comment (rldhont) [reply assets/src/modules/Config.js:254]: same it is a getter

File: assets/src/modules/Config.js (hunk 3 above)
Comment (Gustry) [reply assets/src/modules/Config.js:235]: Indeed, thanks.
TIL

**Comment by rldhont** [comment]
Thanks @Gustry for the review

**Comment by nboisteault** [comment]
@rldhont webpack can't build since this PR.

```
ERROR in ./src/modules/Lizmap.js 30:0-50
Module not found: Error: Can't resolve 'ol-wfs-capabilities' in '/home/nboisteault/3liz/infra/lizmap-web-client/assets/src/modules'
resolve 'ol-wfs-capabilities' in '/home/nboisteault/3liz/infra/lizmap-web-client/assets/src/modules'
  Parsed request is a module
  using description file: /home/nboisteault/3liz/infra/lizmap-web-client/assets/package.json (relative path: ./src/modules)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /home/nboisteault/3liz/infra/lizmap-web-client/assets/src/modules/node_modules doesn't exist or is not a directory
      /home/nboisteault/3liz/infra/lizmap-web-client/assets/src/node_modules doesn't exist or is not a directory
      looking for modules in /home/nboisteault/3liz/infra/lizmap-web-client/assets/node_modules
        single file module
          using description file: /home/nboisteault/3liz/infra/lizmap-web-client/assets/package.json (relative path: ./node_modules/ol-wfs-capabilities)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /home/nboisteault/3liz/infra/lizmap-web-client/assets/node_modules/ol-wfs-capabilities doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /home/nboisteault/3liz/infra/lizmap-web-client/assets/node_modules/ol-wfs-capabilities.js doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /home/nboisteault/3liz/infra/lizmap-web-client/assets/node_modules/ol-wfs-capabilities.json doesn't exist
            .wasm
              Field 'browser' doesn't contain a valid alias configuration
              /home/nboisteault/3liz/infra/lizmap-web-client/assets/node_modules/ol-wfs-capabilities.wasm doesn't exist
        /home/nboisteault/3liz/infra/lizmap-web-client/assets/node_modules/ol-wfs-capabilities doesn't exist
      /home/nboisteault/3liz/infra/lizmap-web-client/node_modules doesn't exist or is not a directory
      /home/nboisteault/3liz/infra/node_modules doesn't exist or is not a directory
      /home/nboisteault/3liz/node_modules doesn't exist or is not a directory
      /home/nboisteault/node_modules doesn't exist or is not a directory
      /home/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
 @ ./src/modules/Globals.js 6:0-33 9:23-29
 @ ./src/index.js 30:0-71 44:28-38 45:37-56

webpack 5.93.0 compiled with 2 errors in 3217 ms

```

**Pull request merged**